contrib/mod_tls.c

```diff
@@ -18340,22 +18210,21 @@ static int tls_ctx_set_ciphers(SSL_CTX *ctx) {
 }
 
 static int tls_ctx_set_crls(SSL_CTX *ctx) {
+  X509_STORE *store;
+
   if (tls_crl_file == NULL &&
       tls_crl_path == NULL) {
     return 0;
   }
 
-  /* Set up the CRL. */
-  tls_crl_store = X509_STORE_new();
-  if (tls_crl_store == NULL) {
-    tls_log("error creating CRL store: %s", tls_get_errors());
+  store = SSL_CTX_get_cert_store(ctx);
+  if (store == NULL) {
+    tls_log("error getting SSL_CTX store: %s", tls_get_errors());
     return -1;
   }
 
   PRIVS_ROOT
-  if (X509_STORE_load_locations(tls_crl_store, tls_crl_file,
-      tls_crl_path) != 1) {
-
+  if (X509_STORE_load_locations(store, tls_crl_file, tls_crl_path) != 1) {
```

Comment:
Here, we load the configured CRLs into the existing `SSL_CTX`.  Much simpler (and better!) this way.